www/js/app.js

```diff
@@ -580,17 +580,49 @@ define(['jquery', 'zimArchiveLoader', 'util', 'uiUtil', 'cookies','abstractFiles
      * Displays the zone to select files from the archive
      */
     function displayFileSelect() {
-        $('#openLocalFiles').show();
-        $('#archiveFiles').on('change', setLocalArchiveFromFileSelect);
+        document.getElementById('openLocalFiles').style.display = 'block';
+        // Make the whole app a drop zone
+        var dropZone = document.getElementById('search-article');
+        dropZone.addEventListener('dragover', function(e) {
+            e.stopPropagation();
+            e.preventDefault();
+            e.dataTransfer.dropEffect = 'copy';
+        }, false);
+        dropZone.addEventListener('drop', setLocalArchiveFromFileList);
+        document.getElementById('archiveFiles').addEventListener('change', setLocalArchiveFromFileSelect);
     }
 
     function setLocalArchiveFromFileList(files) {
+        // Check to see if the files are from file select or from drag-and-drop
+        if (files.dataTransfer) {
+            files.stopPropagation();
+            files.preventDefault();
+            files = files.dataTransfer.files;
+            document.getElementById('openLocalFiles').style.display = 'none';
+            document.getElementById('downloadInstruction').style.display = 'none';
+            document.getElementById('selectorsDisplay').style.display = 'inline';
+        }
+        // Check for usable file types
+        for (var i = files.length; i--;) {
+            if (!/\.(?:zim\w{0,2}|dat|idx|txt)$/i.test(files[i].name)) {
```

Comment:
I seem to remember that the dat, idx and txt extensions were for the old Evopedia archive format. Maybe we only need the .zim* files here?

Reply:
Yes, I was wondering about them. It sounds like those extensions are obsolete. I'll remove.